src/main/java/org/prebid/server/bidder/dmx/DmxBidder.java

```diff
@@ -202,10 +210,11 @@ private Site modifySite(Site site, String updatedPublisherId) {
                         .build();
     }
 
-    private App modifyApp(App app, String updatedPublisherId) {
+    private App modifyApp(App app, Device device, String updatedPublisherId) {
         return app == null
                 ? null
                 : app.toBuilder()
+                        .id(StringUtils.isNotBlank(app.getId()) ? app.getId() : device.getIfa())
```

Comment:
Can device here be null?

Reply:
If `device` is null, then `validateRequest` method would throw an `PreBidException`. I'll add unit test to cover this case

Reply:
As I see validateRequest will not throw ex with some userValues, and null device can be passed forward

Reply:
In real world, we expect that device being `null` is kinda impossible because it holds the info about physical device through which user is interacting with website/app. Anyway, I'll add `device == null` check to `validateRequest`

Reply:
As I know, we have stored requests and AMP, which as I know can not hold any user information

Reply:
Interesting. Seems we can't have null device because of Ortb2ImplicitParametersResolver.populateDevice()